config/routes.rb

```diff
@@ -1,6 +1,13 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  namespace :api, defaults: { format: 'json' } do
+    get 'greeting', to: 'api#index'
+  end
+
+  get '*path', to: 'static#index', constraints: lambda { |req|
+    !req.xhr? && req.format.html?
+  }
```

Comment:
- [ ] None of the random messages is being displayed on the screen when the server is running, this is due to the `No routes matched location "/" `.

Reply:
@od-c0d3r when I type the localhost:3000/api/greetings I can see the json response for the greetings messages however we need to add the button and when it is clicked the random message has to be displayed.

Reply:
@od-c0d3r when I type the `localhost:3000/api/greetings` I can see the JSON response for the greetings messages however we need to add the button and when it is clicked the random message has to be displayed.